python/pyarrow/tests/test_schema.py

```diff
@@ -87,3 +87,10 @@ def test_schema_equals(self):
         del fields[-1]
         sch3 = A.schema(fields)
         assert not sch1.equals(sch3)
+
+
+class TestField(unittest.TestCase):
+    def test_empty_field(self):
+        f = arrow.Field()
+        # This used to segfault on empty field
+        repr(f)
```

Comment:
This is an inherent limitation of Cython extension types, that their constructors are not generally safe to use directly. So when I said "is not supported" I meant, "is not supported as a public API". The other things you cited are "private API" usages. If you look at the other extension types that wrap other Arrow types (DataType, Array, Schema, etc.), you can see they all follow the empty-ctor followed by init-from-shared_ptr